interface/modules/custom_modules/oe-module-faxsms/library/setup_services.php

```diff
@@ -332,11 +349,68 @@ function toggleSetup(id, type = 'single') {
                 <input type="checkbox" class="checkbox" name="dialog" id="dialog" value="1">
             </span>
         </div>
-        <!-- TODO refactor this to have vendor list a global array for future vendor additions -->
         <div class="frame col-12" id="set-service">
             <form id="set_form" name="set_form" class="form" role="form" method="post" action="">
                 <input type="hidden" name="csrf_token_form" value="<?php echo attr(CsrfUtils::collectCsrfToken()); ?>" />
-                <div class="">
+                <div class="form-group">
+                    <?php if (isset($permissions_saved) && $permissions_saved) { ?>
+                    <div class="alert alert-<?php echo attr($permissions_message_type ?? 'success'); ?> text-center alert-dismissible fade show" role="alert">
+                        <strong>
+                            <?php if ($permissions_message_type === 'warning') { ?>
```

Comment:
try:
```
<?php if (($permissions_message_type ?? '') === 'warning') { ?>
```
(it looks like the $permissions_message_type = '' in above code may not always be run)